src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -42,14 +43,21 @@ public R4FhirImportService(IResourceIdentityService resourceIdentityService, Fhi
 
         public override async Task ProcessAsync(ILookupTemplate<IFhirTemplate> config, IMeasurementGroup data, Func<Exception, IMeasurementGroup, Task<bool>> errorConsumer = null)
         {
-            // Get required ids
-            var ids = await ResourceIdentityService.ResolveResourceIdentitiesAsync(data).ConfigureAwait(false);
+            try
+            {
+                // Get required ids
+                var ids = await ResourceIdentityService.ResolveResourceIdentitiesAsync(data).ConfigureAwait(false);
 
-            var grps = _fhirTemplateProcessor.CreateObservationGroups(config, data);
+                var grps = _fhirTemplateProcessor.CreateObservationGroups(config, data);
 
-            foreach (var grp in grps)
+                foreach (var grp in grps)
+                {
+                    _ = await SaveObservationAsync(config, grp, ids).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
             {
-                _ = await SaveObservationAsync(config, grp, ids).ConfigureAwait(false);
+                FhirServiceExceptionProcessor.ProcessException(ex, _logger);

```

Comment:
I think we should retain the previous behaviour and not eat the exception?

Reply:
Just clarifying - do you mean to wrap only the line with `ResolveResourceIdentitiesAsync` in the try-block, so that the other lines are still executed if we do get an exception thrown (for an authorization error) from `ResolveResourceIdentitiesAsync`?

Reply:
To retain the previous behaviour where this function would throw if we encountered an exception, we can just throw it back after processing the exception. Unless we feel like throwing it back doesn't make sense anymore? I'm worried that if we were previously retrying on this exception, then by eating the exception now, we will not be and if that is desired.

Reply:
Oh I see what you mean, thanks for the details. I think you bring up a good point; I'll re-throw while keeping the stack trace.